liquibase-standard/src/main/java/liquibase/diff/output/changelog/DiffToChangeLog.java

```diff
@@ -493,6 +503,10 @@ private List<DatabaseObject> sortObjects(final String type, Collection<DatabaseO
                 }
             } catch (DatabaseException e) {
                 Scope.getCurrentScope().getLog(getClass()).fine("Cannot get object dependencies: " + e.getMessage());
+            } catch (StackOverflowError e) {
+                Scope.getCurrentScope().getLog(getClass()).warning("You have too many or recursive database object dependencies! " +
+                        "Liquibase is going to ignore dependency sorting and resume processing. To skip this message " +
+                        "(and gain a lot of processing time) use flag " + AbstractChangelogCommandStep.SKIP_OBJECT_SORTING.getName(), e);
```

Comment:
Wouldn't you reduce the amount of processing time by skipping sorting?

Reply:
whops, replaced "gain" by "save"